src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1083,7 +1083,7 @@ internal override Cci.IMethodReference Translate(MethodSymbol symbol, Diagnostic
             Debug.Assert(methodSymbol.IsDefinitionOrDistinct());
 
             // Method of anonymous type being translated
-            if (container.IsAnonymousType)
+            if (container?.IsAnonymousType == true)
```

Comment:
As we discussed offline, we are probably injecting these method in the pipeline too early.